File: medplum-java/medplum-model/src/main/java/com/fhir/MedicationKnowledge_Dosage.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedicationKnowledge_Dosage.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface MedicationKnowledge_Dosage {
  CodeableConcept type();

  Optional<String> id();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  List<Dosage> dosage();

  static ImmutableMedicationKnowledge_Dosage.TypeBuildStage builder() {
    return ImmutableMedicationKnowledge_Dosage.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/MedicationKnowledge_Dosage.ts
import { CodeableConcept } from "./CodeableConcept";
import { Dosage } from "./Dosage";
import { Extension } from "./Extension";

export interface MedicationKnowledge_Dosage {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    type: CodeableConcept;
    dosage: Dosage[];
}
